readthedocs/restapi/views/model_views.py

```diff
@@ -45,11 +45,13 @@ class PlainTextRenderer(BaseRenderer):
     format = 'txt'
 
     def render(self, data, accepted_media_type=None, renderer_context=None):
-        kwargs = renderer_context.get('kwargs', {})
-        if kwargs.get('format', self.format) == self.format:
-            return data.encode(self.charset)
-        else:
-            raise Http404
+        response = renderer_context.get('response')
```

Comment:
If `render_context` is `None` (the default) this will fail.

Besides, I don't understand how it works. Why sometime we can have the `data` already generated and why other times we need to `render_to_string`? I'm not following this.

Reply:
When the response status is 200, we have the data from the model, but when the status is a error, we have a single dict {"detail": "explanation"'}. I think this is done in this way because the renderer only represent the data in another way, but here we are using the renderer for a specific case.

I'm not sure about the `None` value, but looking at the code of others renderes, this is a common pattern `renderer_context = renderer_context or {}`

Reply:
Maybe here is better to use `response.exception` as https://github.com/encode/django-rest-framework/blob/master/rest_framework/renderers.py#L165

instead of checking by the status code. Seems more readable.